azure/src/main/java/org/apache/iceberg/azure/adlsv2/ADLSLocation.java

```diff
@@ -18,24 +18,34 @@
  */
 package org.apache.iceberg.azure.adlsv2;
 
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.util.Optional;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import org.apache.iceberg.exceptions.ValidationException;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 
 /**
- * This class represents a fully qualified location in Azure expressed as a URI.
+ * This class represents a fully qualified location in Azure Data Lake Storage, expressed as a URI.
  *
  * <p>Locations follow the conventions used by Hadoop's Azure support, i.e.
  *
- * <pre>{@code abfs[s]://[<container>@]<storage account host>/<file path>}</pre>
+ * <pre>{@code abfs[s]://[<container>@]<storageAccount>.dfs.core.windows.net/<path>}</pre>
```

Comment:
Is this a change? or is Storage Account Host equivalent to `storageAccount.dfs.core.windows.net`

Reply:
I think `storageAccount` is more commonly used than `storageHost`. ([ref](https://learn.microsoft.com/en-us/azure/storage/common/storage-account-overview#standard-endpoints))

Reply:
I see the ambiguity here so I changed the variable to use `storageEndpoint` to clarify.  Previously we were using the `storageAccount` variable to store the entire endpoint's hostname and not the storage account's name